network/crl.go

```diff
@@ -395,57 +395,57 @@ func checkCertWithCRL(cert *x509.Certificate, cacheItem *CRLCacheItem) error {
 		}
 	}
 
-	revokedCertificate, ok := cacheItem.RevokedCertificates[cert.SerialNumber]
-	if ok {
-		log.WithField("extensions", revokedCertificate.Extensions).Debugln("Revoked certificate extensions")
-		for _, extension := range revokedCertificate.Extensions {
-			// One can use https://www.alvestrand.no/objectid/top.html to convert string OIDs (like id-ce-keyUsage)
-			// into numerical format (like 2.5.29.15), though RFC also allows easy reconstruction of OIDs.
-
-			// Some of these extensions values may be ignored, but we still have to
-			// include them (their OIDs) here so that we're not blindly ignoring them.
-			// Convert to string to be able to use switch.
-			switch extension.Id.String() {
-			// As per RFC 5280 section 4.2, aplications MUST recognize following extensions:
-			case "2.5.29.15":
-				// section 4.2.1.3, id-ce-keyUsage
-			case "2.5.29.32":
-				// section 4.2.1.4, id-ce-certificatePolicies
-			case "2.5.29.17":
-				// section 4.2.1.6, id-ce-subjectAltName
-			case "2.5.29.19":
-				// section 4.2.1.9, id-ce-basicConstraints
-			case "2.5.29.30":
-				// section 4.2.1.10, id-ce-nameConstraints
-			case "2.5.29.36":
-				// section 4.2.1.11, id-ce-policyConstraints
-			case "2.5.29.37":
-				// section 4.2.1.12, id-ce-extKeyUsage
-			case "2.5.29.54":
-				// section 4.2.1.14, id-ce-inhibitAnyPolicy
-
-			// As per RFC 5280 section 4.2, aplications SHOULD recognize following extensions:
-			case "2.5.29.35":
-				// section 4.2.1.1, id-ce-authorityKeyIdentifier
-			case "2.5.29.14":
-				// section 4.2.1.2, id-ce-subjectKeyIdentifier
-			case "2.5.29.33":
-				// section 4.2.1.5, id-ce-policyMappings
-
-			default:
-				if extension.Critical {
-					log.WithField("oid", extension.Id.String()).Warnln("CRL: Unable to process critical extension with unknown Object ID")
-					return ErrUnknownCRLExtensionOID
-				}
-				log.WithField("oid", extension.Id.String()).Debugln("CRL: Unable to process non-critical extension with unknown Object ID")
+	revokedCertificate, ok := cacheItem.RevokedCertificates[cert.SerialNumber.Text(16)]
```

Comment:
if we use same integer in several places, lets use constant for that to avoid remembering in future changes

Reply:
BTW, you also proposed using base62 for this to have smaller strings. But seems like `.Text()` has [special implementation for power-of-two bases](https://cs.opensource.google/go/go/+/refs/tags/go1.17.6:src/math/big/natconv.go;l=282;drc=refs%2Ftags%2Fgo1.17.6), which makes me think this should work faster with no division involved. Then I will make the constant and set it to 32. A bit longer that base62, but hashing should be faster.